### PR TITLE
Added code for reaction rate in a diffusion-reaction system

### PR DESCRIPTION
**Pull request by eddieqiao23**
l2 relative error for k: 0.024211755
l2 relative error for u: 0.0011301273638472755

### CODE REVIEW

File: docs/demos/pinn_inverse.rst [hunk 1]
```diff
@@ -23,6 +23,8 @@ PDEs
    pinn_inverse/reaction.inverse
    pinn_inverse/elliptic.inverse.field
 
+- `Inverse problem for the reaction rate in a diffusion-reaction system <https://github.com/lululxvi/deepxde/blob/master/examples/pinn_inverse/diffusion_reaction_rate.py>`_
```
Comment (lululxvi) [inline docs/demos/pinn_inverse.rst:26]: Inverse problem for a diffusion-reaction system

File: examples/pinn_inverse/diffusion_reaction_rate.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
+"""Backend supported: tensorflow.compat.v1
+
+Implementation of the space-dependent reaction rate in a diffusion-reaction system in paper https://arxiv.org/abs/2111.02801.
+"""
+import deepxde as dde 
+import numpy as np
+from scipy.integrate import solve_bvp
+import matplotlib.pyplot as plt
+from deepxde.backend import tf
```
Comment (lululxvi) [inline examples/pinn_inverse/diffusion_reaction_rate.py:9]: Reorder
```
import deepxde as dde
import matplotlib.pyplot as plt
import numpy as np
from deepxde.backend import tf
from scipy.integrate import solve_bvp
```

**Comment by lululxvi** [comment]
Reformat the code using black

File: examples/pinn_inverse/diffusion_reaction_rate.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    geom,
+    pde,
+    bcs=[bc, observe_u],
+    num_domain=50,
+    num_boundary=8,
+    train_distribution="uniform",
+    num_test=1000,
+)
+
+net = dde.maps.PFNN([1, [20, 20], [20, 20], 2], "tanh", "Glorot uniform")
```
Comment (lululxvi) [inline examples/pinn_inverse/diffusion_reaction_rate.py:64]: change `dde.maps.PFNN` to `dde.nn.PFNN`

File: examples/pinn_inverse/diffusion_reaction_rate.py [hunk 4]
```diff
@@ -0,0 +1,90 @@
+"""Backend supported: tensorflow.compat.v1
```
Comment (lululxvi) [inline examples/pinn_inverse/diffusion_reaction_rate.py:1]: How about tf2 and Pytorch?

**Comment by eddieqiao23** [comment]
It shows "AttributeError: module 'deepxde.nn.pytorch' has no attribute 'PFNN'" and the same for tf2

File: examples/pinn_inverse/diffusion_reaction_rate.py (hunk 4 above)
Comment (eddieqiao23) [reply examples/pinn_inverse/diffusion_reaction_rate.py:1]: It shows "AttributeError: module 'deepxde.nn.pytorch' has no attribute 'PFNN'" and the same for tf2

**Pull request merged**